sycl/include/CL/sycl/detail/property_base.hpp

```diff
@@ -0,0 +1,27 @@
+//==------- property_base.hpp --- SYCL property base class -----------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <CL/sycl/detail/defines.hpp>
+
+__SYCL_INLINE_NAMESPACE(cl) {
+namespace sycl {
+namespace property {
+namespace detail {
+
+// Property base class
+class property_base {
+public:
+  ~property_base() = default;
```

Comment:
It should be virtual if there are derivatives of `property_base` class.

Reply:
You're absolutely right! I've fixed this mistake.